projects/igniteui-angular/src/lib/tree/tree-node/tree-node.component.ts

```diff
@@ -0,0 +1,355 @@
+    public children: QueryList<IgxTreeNode<any>>;
+
+    // TODO: Expose in public API instead of `children` query list
+    /**
+     * Return the child nodes of the node (if any)
+     *
+     * @remark
+     * Returns `null` if node does not have children
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const children: IgxTreeNode<any>[] = node.children;
+     * ```
+     */
+    public get _children(): IgxTreeNode<any>[] {
+        return this.children?.length ? this.children.toArray() : null;
+    }
+
+    @ViewChild('defaultSelect', { read: TemplateRef, static: true })
+    private _defaultSelectMarker: TemplateRef<any>;
+
+    @ViewChild('defaultIndicator', { read: TemplateRef, static: true })
+    private _defaultExpandIndicatorTemplate: TemplateRef<any>;
+
+    @ViewChild('childrenContainer', { read: ElementRef })
+    private childrenContainer: ElementRef;
+
+    // TODO: this should probably be dropped from the API
+    /**
+     * The unique ID of the node
+     */
+    public id = `igxTreeNode_${nodeId++}`;
+
+    constructor(
+        @Inject(IGX_TREE_COMPONENT) public tree: IgxTree,
+        protected selectionService: IgxTreeSelectionService,
+        protected treeService: IgxTreeService,
+        protected cdr: ChangeDetectorRef,
+        protected builder: AnimationBuilder,
+        private element: ElementRef<HTMLElement>,
+        @Optional() @SkipSelf() @Inject(IGX_TREE_NODE_COMPONENT) public parentNode: IgxTreeNode<any>
+    ) {
+        super(builder);
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get showSelectors() {
+        return this.tree.selection !== IGX_TREE_SELECTION_TYPE.None;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get indeterminate(): boolean {
+        return this.selectionService.isNodeIndeterminate(this);
+    }
+
+    /** The depth of the node, relative to the root
+     *
+     * ```html
+     * <igx-tree>
+     *  ...
+     *  <igx-tree-node #node>
+     *      My level is {{ node.level }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[12])[0];
+     * const level: number = node.level;
+     * ```
+     */
+    public get level(): number {
+        return this.parentNode ? this.parentNode.level + 1 : 0;
+    }
+
+    /** Get/set whether the node is selected. Supporst two-way binding.
+     *
+     * ```html
+     * <igx-tree>
+     *  ...
+     *  <igx-tree-node *ngFor="let node of data" [(selected)]="node.selected">
+     *      {{ node.label }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const selected = node.selected;
+     * node.selected = true;
+     * ```
+     */
+    @Input()
+    public get selected(): boolean {
```

Comment:
Also you need `selectedChange` for `[(selected)]` binding.

Reply:
One was implemented w/ the selection service, it's at the start of the class definition (to avoid making lint angry)